### PR TITLE
Fix `Too many requests` error in github actions test

### LINKED ISSUE #19597: `TestAWS` flakiness (`Too many requests` error in tsh tests)
https://github.com/gravitational/teleport/actions/runs/3753666381/jobs/6377134366

```
make: *** [test-go] Error 1
    tsh_test.go:2540: 
        	Error Trace:	/__w/teleport/teleport/tool/tsh/tsh_test.go:2540
        	            				/__w/teleport/teleport/tool/tsh/aws_test.go:46
        	Error:      	Received unexpected error:
        	            	Too many requests
        	            		
        	Test:       	TestAWS
--- FAIL: TestAWS (0.04s)
```

"Too many requests" looks suspicious - does the test actually make calls to AWS services?

### PR DESCRIPTION
**Pull request by greedy52**
Fixes #19597 

Failure
```
    tsh_test.go:3690: 
        	Error Trace:	/__w/teleport/teleport/tool/tsh/tsh_test.go:2547
        	            				/__w/teleport/teleport/tool/tsh/tsh_test.go:3690
        	Error:      	Received unexpected error:
        	            	Too many requests
```
Trace
```
Original Error: *trace.LimitExceededError 
Stack Trace:
	/__w/teleport/teleport/lib/cloud/azure/errors.go:56 github.com/gravitational/teleport/lib/cloud/azure.parseMetadataClientError
	/__w/teleport/teleport/lib/cloud/azure/imds.go:131 github.com/gravitational/teleport/lib/cloud/azure.(*InstanceMetadataClient).getRawMetadata
	/__w/teleport/teleport/lib/cloud/azure/imds.go:171 github.com/gravitational/teleport/lib/cloud/azure.(*InstanceMetadataClient).GetTags
	/__w/teleport/teleport/lib/cloud/azure/imds.go:189 github.com/gravitational/teleport/lib/cloud/azure.(*InstanceMetadataClient).GetHostname
	/__w/teleport/teleport/lib/service/service.go:890 github.com/gravitational/teleport/lib/service.NewTeleport
	/__w/teleport/teleport/tool/tsh/tsh_test.go:2543 github.com/gravitational/teleport/tool/tsh.makeTestServers
	/__w/teleport/teleport/tool/tsh/tsh_test.go:3812 github.com/gravitational/teleport/tool/tsh.TestExportingTraces.func4
	/opt/go/src/testing/testing.go:1446 testing.tRunner
	/opt/go/src/runtime/asm_amd64.s:1594 runtime.goexit
User Message: Too many requests
	] tsh/tsh_test.go:2545
```

This change sets `DisabledIMDSClient` for new Teleport service in unit tests and integration tests, e.g:
```
config.InstanceMetadataClient = cloud.NewDisabledIMDSClient()
```

### CODE REVIEW

File: lib/service/service.go [hunk 1]
```diff
@@ -886,7 +892,7 @@ func NewTeleport(cfg *Config, opts ...NewTeleportOption) (*TeleportProcess, erro
 		}
 	}
 
-	if imClient != nil {
+	if imClient != nil && imClient.GetType() != types.InstanceMetadataTypeDisabled {
```
Comment (greedy52) [inline lib/service/service.go:895]: skipping if type is `types.InstanceMetadataTypeDisabled`

**Review by r0mant — COMMENTED** [review]
@greedy52 Maybe make it a part of Teleport config instead of a functional option?

**Comment by greedy52** [comment]
> @greedy52 Maybe make it a part of Teleport config instead of a functional option?

Yes, let me try that.

**Review by r0mant — APPROVED** [review]
Cool, I think this is a little cleaner than having to pass the functional argument everywhere :+1:

**Review by jakule — APPROVED** [review]
@greedy52 Thanks for fixing that.

**Pull request merged**